tests/consistent.rs

```diff
@@ -0,0 +1,213 @@
+
+        ("pikevm_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_utf8bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bytes(true)
+                    .build()
+                    .map(|exec| exec.into_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    let bytes_backends = vec![
+        ("bounded_backtracking_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .bounded_backtracking()
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("pikevm_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .nfa()
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+
+        ("default_bytes_re",
+            try!(ExecBuilder::new(re)
+                    .only_utf8(false)
+                    .build()
+                    .map(|exec| exec.into_byte_regex())
+                    .map_err(|err| format!("{}", err)))),
+    ];
+
+    Ok(try!(string_checker::check_backends(&standard_backends))
+       + try!(string_checker::check_backends(&utf8bytes_backends))
+       + try!(bytes_checker::check_backends(&bytes_backends)))
+}
+
+//
+// A consistency checker parameterized by the input type (&str or &[u8]).
+//
+
+macro_rules! checker {
+    ($module_name:ident, $regex_type:path, $mk_input:expr) => {
+        mod $module_name {
+            // use $regex_type;
+            use super::quickcheck;
+            use super::quickcheck::{TestResult, Arbitrary};
+
+pub fn check_backends(backends: &[(&str, $regex_type)]) -> Result<u64, String> {
+    let mut total_passed = 0;
+    for regex in backends[1..].iter() {
+        total_passed += try!(quickcheck_regex_eq(&backends[0], regex));
+    }
+
+    Ok(total_passed)
+}
+
+fn quickcheck_regex_eq(
+    &(name1, ref re1): &(&str, $regex_type),
+    &(name2, ref re2): &(&str, $regex_type),
+) -> Result<u64, String> {
+    quickcheck::QuickCheck::new()
+        .quicktest(RegexEqualityTest::new(re1.clone(), re2.clone()))
+        .map_err(|err|
+            format!("{}(/{}/) and {}(/{}/) are inconsistent.\nQuickCheck Err: {:?}",
+                    name1, re1, name2, re2, err))
+}
+
+struct RegexEqualityTest {
+    re1: $regex_type,
+    re2: $regex_type,
+}
+impl RegexEqualityTest {
+    fn new(re1: $regex_type, re2: $regex_type) -> Self {
+        RegexEqualityTest {
+            re1: re1,
+            re2: re2,
+        }
+    }
+}
+
+impl quickcheck::Testable for RegexEqualityTest {
```

Comment:
This is kind of weird? What made you implement `Testable` explicitly here?

Reply:
I ran into some trouble with the borrow checker when I tried to use a closure over the two regex in question. I think the closures tried to take ownership of the regex, when I wanted to copy them instead.